common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -188,89 +195,69 @@ public ByteString newInstance(byte[] value, int start, int length, boolean copy)
      * {@link CharSequence}s into the same {@link TextHeaders}.
      */
     public static int caseInsensitiveHashCode(CharSequence value) {
-        if (value instanceof AsciiString) {
-            try {
-                ByteProcessor processor = new ByteProcessor() {
-                    private int hash;
-                    @Override
-                    public boolean process(byte value) throws Exception {
-                        hash = hash * HASH_CODE_PRIME ^ toLowerCase(value) & HASH_CODE_PRIME;
-                        return true;
-                    }
-
-                    @Override
-                    public int hashCode() {
-                        return hash;
-                    }
-                };
-                ((AsciiString) value).forEachByte(processor);
-                return processor.hashCode();
-            } catch (Exception e) {
-                PlatformDependent.throwException(e);
+        if (value == null) {
+            return CASE_INSENSITIVE_HASHER.emptyHashValue();
+        }
+        if (value.getClass() == AsciiString.class) {
+            return ((AsciiString) value).hashCodeCaseInsensitive();
+        }
+        if (value instanceof CharBuffer) {
+            CharBuffer buffer = (CharBuffer) value;
+            if (buffer.hasArray() && !buffer.isReadOnly()) {
+                final int startPos = buffer.arrayOffset() + buffer.position();
+                return CASE_INSENSITIVE_HASHER.hashCodeAsBytes(
+                        buffer.array(), startPos, startPos + buffer.remaining());
             }
         }
 
-        int hash = 0;
-        final int end = value.length();
-        for (int i = 0; i < end; i++) {
-            hash = hash * HASH_CODE_PRIME ^ toLowerCase(value.charAt(i)) & HASH_CODE_PRIME;
-        }
-        return hash;
+        return CASE_INSENSITIVE_HASHER.hashCodeAsBytes(value, 0, value.length());
     }
 
     /**
      * Returns {@code true} if both {@link CharSequence}'s are equals when ignore the case. This only supports 8-bit
      * ASCII.
      */
-    public static boolean equalsIgnoreCase(CharSequence a, CharSequence b) {
+    public static boolean contentEqualsIgnoreCase(CharSequence a, CharSequence b) {
```

Comment:
@nmittler - Or also see [ByteString.equals(..)](https://github.com/netty/netty/blob/master/common/src/main/java/io/netty/util/ByteString.java#L661) for an example of types must be compatible for `equals` to be `true`.